core/templates/dev/head/pages/admin/roles_tab/RoleGraphDirective.js

```diff
@@ -0,0 +1,103 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for displaying Role graph.
+ */
+
+oppia.directive('roleGraph', [
+  'UrlInterpolationService', function(UrlInterpolationService) {
+    return {
+      restrict: 'E',
+      scope: {
+        // An object with these keys:
+        //  - 'nodes': An object whose keys are node ids and whose values are
+        //             node labels
+        //  - 'links': A list of objects with keys:
+        //            'source': id of source node
+        //            'target': id of target node
+        //            'linkProperty': property of link which determines how
+        //              it is styled (styles in linkPropertyMapping). If
+        //              linkProperty or corresponding linkPropertyMatching
+        //              is undefined, link style defaults to the gray arrow.
+        //  - 'initStateId': The initial state id
+        //  - 'finalStateIds': The list of ids corresponding to terminal states
+        //             (i.e., those whose interactions are terminal).
```

Comment:
Probably shouldn't mention interactions in this comment, since they're specifically an exploration thing.